File: svws-webclient/core/src/asd/validate/lehrer/ValidatorLehrerStammdatenNachnameVorhanden.ts
import { LehrerStammdaten } from '../../../asd/data/lehrer/LehrerStammdaten';
import { Class } from '../../../java/lang/Class';
import { JavaString } from '../../../java/lang/JavaString';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorLehrerStammdatenNachnameVorhanden extends Validator<LehrerStammdaten> {


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param daten     die Daten des Validators
	 * @param kontext   der Kontext des Validators
	 */
	public constructor(daten : LehrerStammdaten, kontext : ValidatorKontext) {
		super(daten, kontext);
	}

	protected pruefe() : boolean {
		const nachname : string | null = this.daten().nachname;
		if ((nachname === null) || (nachname.length === 0)) {
			this.addFehler("Kein Wert im Feld 'nachname'.");
			return false;
		}
		if (JavaString.isBlank(nachname.trim())) {
			this.addFehler("Das Feld 'nachname' darf nicht nur aus Leerzeichen bestehen.");
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.lehrer.ValidatorLehrerStammdatenNachnameVorhanden';
	}

	isTranspiledInstanceOf(name : string): boolean {
		return ['de.svws_nrw.asd.validate.lehrer.ValidatorLehrerStammdatenNachnameVorhanden', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static class = new Class<ValidatorLehrerStammdatenNachnameVorhanden>('de.svws_nrw.asd.validate.lehrer.ValidatorLehrerStammdatenNachnameVorhanden');

}

export function cast_de_svws_nrw_asd_validate_lehrer_ValidatorLehrerStammdatenNachnameVorhanden(obj : unknown) : ValidatorLehrerStammdatenNachnameVorhanden {
	return obj as ValidatorLehrerStammdatenNachnameVorhanden;
}

File: svws-asd/src/main/java/de/svws_nrw/asd/validate/lehrer/ValidatorLehrerStammdatenNachnameVorhanden.java
package de.svws_nrw.asd.validate.lehrer;

import de.svws_nrw.asd.data.lehrer.LehrerStammdaten;
import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import jakarta.validation.constraints.NotNull;

/**
 * Dieser Validator führt eine Statistikprüfung aus, ob ein Nachname bei den Stammdaten
 * eines Lehrers einer Schule angegeben wurde oder nicht.
 */
public final class ValidatorLehrerStammdatenNachnameVorhanden extends Validator {

	/** Die Lehrer-Stammdaten */
	private final @NotNull LehrerStammdaten daten;

	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param daten     die Daten des Validators
	 * @param kontext   der Kontext des Validators
	 */
	public ValidatorLehrerStammdatenNachnameVorhanden(final @NotNull LehrerStammdaten daten,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		this.daten = daten;
	}

	@Override
	protected boolean pruefe() {
		final String nachname = daten.nachname;
		if ((nachname == null) || (nachname.length() == 0)) {
			addFehler("Kein Wert im Feld 'nachname'.");
			return false;
		}
		if (nachname.trim().isBlank()) {
			addFehler("Das Feld 'nachname' darf nicht nur aus Leerzeichen bestehen.");
			return false;
		}
		return true;
	}

}
